### PR TITLE
Updates build documentation

### PR DESCRIPTION
**Pull request by JanuszL**
- updates the information about the availability
  of the python bindings for the Jeston aarch64 build
- removes the info about QNX support
- removes supported dependencies versions and replaces links to
  https://github.com/NVIDIA/DALI_deps

#### Why we need this PR?
*Pick one, remove the rest*
- It updates build documentation

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     updates the information about the availability of the python bindings for the Jeston aarch64 build
     removes the info about QNX support
     removes supported dependencies versions and replaces links to https://github.com/NVIDIA/DALI_deps
 - Affected modules and functionalities:
     compilation section of the documentation
 - Key points relevant for the review:
     wording
 - Validation and testing:
     visual inspection
 - Documentation (including examples):
     compilation section of the documentation was updated

**JIRA TASK**: *[NA]*

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

File: docs/compilation.rst [hunk 1]
```diff
@@ -115,30 +115,32 @@ Bare Metal build
 Prerequisites
 ^^^^^^^^^^^^^
 
+For the recommended dependencies versions please check https://github.com/NVIDIA/DALI_deps.
```
Comment (mzient) [inline docs/compilation.rst:118]:
```suggestion
For the recommended dependency versions please check https://github.com/NVIDIA/DALI_deps.
```

File: docs/compilation.rst [hunk 2]
```diff
@@ -422,14 +424,14 @@ Following CMake parameters could be helpful in setting the right paths:
...
-------------------------------------------
+Cross-compiling for aarch64 Jetson Linux (Docker)
+-------------------------------------------------
 
 .. note::
 
   Support for aarch64 Linux platform is experimental. Some of the features are available only for
-  x86-64 target and they are turned off in this build. There is no support for DALI Python library
-  on aarch64 yet. Some Operators may not work as intended due to x86-64 specific implementations.
+  x86-64 target and they are turned off in this build. Some Operators may not work as intended due
```
Comment (mzient) [inline docs/compilation.rst:433]:
```suggestion
  x86-64 target and they are turned off in this build. Some operators may not work as intended due
```

Comment (mzient) [reply docs/compilation.rst:433]: I have mixed feelings about this sentence. Why should we even say such things?

Comment (jantonguirao) [reply docs/compilation.rst:433]: I'd also vote for removing this

File: docs/compilation.rst [hunk 3]
```diff
@@ -458,33 +460,6 @@ From the root of the DALI source tree
...
-debian package into the ``qnx`` directory you created in the DALI tree.
-
-Build the aarch64 Build Container
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-.. code-block:: bash
-
-    docker build -t nvidia/dali:tools_aarch64-qnx -f docker/Dockerfile.cuda_qnx.deps .
-    docker build -t nvidia/dali:builder_aarch64-qnx --build-arg "QNX_CUDA_TOOL_IMAGE_NAME=nvidia/dali:tools_aarch64-qnx" -f docker/Dockerfile.build.aarch64-qnx .
-
```
Comment (jantonguirao) [inline docs/compilation.rst:487]: The "Compile" section below still has QNX instructions. Is that intended?

File: docs/compilation.rst (hunk 1 above)
Comment (JanuszL) [reply docs/compilation.rst:118]: Done

File: docs/compilation.rst (hunk 2 above)
Comment (JanuszL) [reply docs/compilation.rst]: Done

File: docs/compilation.rst (hunk 3 above)
Comment (JanuszL) [reply docs/compilation.rst:487]: Not really. How about now?

**Review by mzient — APPROVED** [review]

**Review by jantonguirao — APPROVED** [review]

**Comment by JanuszL** [comment]
!build

**Pull request merged**